datafusion/expr/src/logical_plan/extension.rs

```diff
@@ -57,7 +57,7 @@ pub trait UserDefinedLogicalNode: fmt::Debug + Send + Sync {
     fn schema(&self) -> &DFSchemaRef;
 
     /// Perform check of invariants for the extension node.
-    fn check_invariants(&self, check: InvariantLevel, plan: &LogicalPlan) -> Result<()>;
+    fn check_invariants(&self, check: InvariantLevel) -> Result<()>;
```

Comment:
I think technically this is an API change. Given this is a public API I worry that this change might break downstream users and there is no other way to get at the plan